tests/ui/traits/negative-bounds/opaque-type-unsatisfied-bound.rs

```diff
@@ -0,0 +1,23 @@
+// compile-flags: -Znext-solver
+
+#![feature(negative_bounds, negative_impls)]
+
+trait Trait {}
+impl !Trait for () {}
+
+fn produce() -> impl !Trait {}
+fn consume(_: impl Trait) {}
+
+fn main() {
+    consume(produce()); //~ ERROR the trait bound `impl !Trait: Trait` is not satisfied
+}
+
+fn weird0() -> impl Sized + !Sized {}
+//~^ ERROR mismatched types
+//~| ERROR type mismatch resolving `() == impl !Sized`
+fn weird1() -> impl Sized + !Sized {}
+//~^ ERROR mismatched types
+//~| ERROR type mismatch resolving `() == impl !Sized`
+fn weird2() -> impl !Sized + Sized {}
```

Comment:
We elaborate `!Sized` to `!Sized + Sized` which I guess we should change?
Currently, I “unelaborate” any `Sized` bounds again even if we have a `!Sized` bound, too.

Reply:
`!Sized` should opt out implicit sized bounds like `?Sized`